charms/kubernetes_snaps.py

```diff
@@ -148,35 +149,22 @@ def configure_apiserver(
     if external_cloud_provider.has_xcp:
         api_opts["cloud-provider"] = "external"
 
+    # TODO: Check if cloud-config is still required
+    # api_cloud_config_path = cloud_config_path("kube-apiserver")
     if external_cloud_provider.name == "gce":
         # TODO: We need a charm for external-cloud-provider for GCP
         # based on https://github.com/kubernetes/cloud-provider-gcp/
+        api_opts["cloud-provider"] = "gce"
+        # api_opts["cloud-config"] = str(api_cloud_config_path)
```

Comment:
Sadly, we can't do this.  If we set that it loses the external provider.  We're gonna have to make an out-of-tree charm for GCE
```suggestion
```